services/generic/thermostat.md

```diff
@@ -1,56 +1,94 @@
+| `sup_setpoints` | str_array | `["heat", "cool"]`                | List of supported setpoints for which a value can be set. Usually a sub-set of `sup_modes`. See the definition of [`setpoint`](#definitions) for more information. |
+| `sup_states`    | str_array | `["idle", "heat", "cool"]`        | List of supported states, see the definition of [`state`](#definitions) for well-defined examples.                                                                 |
+
+> A device may define their own modes and setpoints outside the well-defined lists contained in the definitions section.
+> However, it is recommended to use the well-defined values whenever possible.
 
-Type | Interface               | Value type | Description
------|-------------------------|------------|------------------
-in   | cmd.mode.get_report     | null       |
-in   | cmd.mode.set            | string     | Set thermostat mode.
-out  | evt.mode.report         | string     |
--|||
-in   | cmd.setpoint.get_report | string     | value is a set-point type
-in   | cmd.setpoint.set        | str_map    | val = {"type":"heat", "temp":"21.5", "unit":"C"}
-out  | evt.setpoint.report     | str_map    | val = {"type":"heat", "temp":"21.5", "unit":"C"}
--|||
-in   | cmd.state.get_report    | null       |
-out  | evt.state.report        | string     | Reports operational state.
+## Definitions
 
-#### Service props
+* `setpoint_map` is a string map with the following structure:
 
-Name            | Value example                                                                  | Description
-----------------|--------------------------------------------------------------------------------|-------------
-`sup_modes`     | off, heat, cool                                                                | Supported modes.
-`sup_setpoints` | heat, cool                                                                     | Supported set-points.
-`sup_states`    | idle, heat, cool, idle, heat, cool, fan_only, pending_heat, pending_cool, vent |
+| Field | Type   | Example  | Description                                                                           |
+|-------|--------|----------|---------------------------------------------------------------------------------------|
+| type  | string | `"heat"` | One of the `mode` values declared in [`sup_setpoints`](#service-properties) property. |
+| temp  | string | `"21.5"` | Setpoint value in the provided unit.                                                  |
+| unit  | string | `"C"`    | Setpoint unit.                                                                        |
 
-Modes: off, heat, cool, auto, aux_heat, resume, fan, furnace, dry_air, moist_air, auto_changeover, energy_heat, energy_cool, away.
+* `mode` is a mode of operation of the device, well-defined modes include: `off`, `heat`, `cool`, `aux_heat`, `energy_heat`, `energy_cool`, `fan`, `fan_only`, `auto`,
```

Comment:
Move desription about mode here: "A modes may use one or more setpoints eg. `auto` mode utilizes both `heat` and `cool` setpoints.

I woudl not use words corresponding as its not precise.

Reply:
Yeah, this sounds a bit easier to understands.